motionplan/motionPlanner.go

```diff
@@ -343,14 +374,20 @@ func (mp *planner) smoothPath(ctx context.Context, path []node) []node {
 // getSolutions will initiate an IK solver for the given position and seed, collect solutions, and score them by constraints.
 // If maxSolutions is positive, once that many solutions have been collected, the solver will terminate and return that many solutions.
 // If minScore is positive, if a solution scoring below that amount is found, the solver will terminate and return that one solution.
-func (mp *planner) getSolutions(ctx context.Context, seed map[string][]referenceframe.Input) ([]node, error) {
+func (mp *planner) getSolutions(ctx context.Context, seed map[string][]referenceframe.Input, metric ik.StateFSMetric) ([]node, error) {
 	// Linter doesn't properly handle loop labels
 	nSolutions := mp.planOpts.MaxSolutions
 	if nSolutions == 0 {
 		nSolutions = defaultSolutionsToSeed
 	}
+	if len(seed) == 0 {
+		// If no seed is passed, generate one randomly
+		for _, frameName := range mp.fs.FrameNames() {
+			seed[frameName] = referenceframe.RandomFrameInputs(mp.fs.Frame(frameName), mp.randseed)
```

Comment:
A nil map is defined as having length zero; this will not panic, and moreover, the linter will not allow a simultaneous nil and len==0 check. A map *can* have len==0 while being non-nil.

As we can now plan from a set of poses to another set of poses, randomly generating the IK seed is reasonable IMO.

Reply:
I think this mini example shows it can panic in this case https://go.dev/play/p/fKTfYC10p5Z

Reply:
Oof, good point. Will re-add that in and massage the linter to make it not complain